### PR TITLE
New student deletion flow

### PR DESCRIPTION
**Pull request by maddiedierker**
New UI flow for student account deletion ([spec](https://docs.google.com/document/d/1dTaA7F7y-ok6fJXzbtWCkJ4aNrk66r78y-gu9wTKr8I/edit?usp=sharing)).

After successfully submitting, the user is logged out and routed to "/".

### Examples
screen shot 2018-07-11 at 3 31 55 pm

When submitted with the wrong password:
screen shot 2018-07-11 at 3 32 10 pm

On an unexpected error:
screen shot 2018-07-11 at 3 33 47 pm

### Follow-up Work
- Add to storybook

### CODE REVIEW

File: apps/src/sites/studio/pages/devise/registrations/edit.js [hunk 1]
```diff
@@ -49,6 +53,15 @@ $(document).ready(() => {
     );
   }
 
+  if (experiments.isEnabled(experiments.ACCOUNT_DELETION_NEW_FLOW)) {
+    const deleteAccountMountPoint = document.getElementById('delete-account');
+    // Replace deleteAccountMountPoint Rails contents with DeleteAccount component.
+    ReactDOM.render(
```
Comment (islemaster) [inline apps/src/sites/studio/pages/devise/registrations/edit.js]: I think this needs to check if the mount point actually exists - it's possible it won't be rendered if the user doesn't have permission to delete their own account.

File: dashboard/app/views/devise/registrations/edit.html.haml [hunk 2]
```diff
@@ -177,16 +177,20 @@
       %span#change-user-type-status
       %button.btn#change-user-type-button{disabled: true}= t('signup_form.user_type_button')
 
-%hr.danger.account-page-section-break
-%h2.danger= t('user.delete')
```
Comment (islemaster) [inline dashboard/app/views/devise/registrations/edit.html.haml:181]: Do we no longer show the horizontal rule and header for the Delete Account section in the case where the user cannot delete their own account?

Comment (maddiedierker) [reply dashboard/app/views/devise/registrations/edit.html.haml:181]: 😱 thanks for catching that! i think i resolved a merge conflict incorrectly. i'll add that back

File: apps/src/sites/studio/pages/devise/registrations/edit.js (hunk 1 above)
Comment (maddiedierker) [reply apps/src/sites/studio/pages/devise/registrations/edit.js]: you're right! i'll add that check.

**Pull request merged**